src/models/scan.rs

```diff
@@ -22,25 +35,139 @@ impl Scan {
     /// depending on the error type.
     /// 3. Find the labware by its barcode `labware_barcode`.
     /// 4. If the labware exists, return it. If it doesn't exist, create the labware in the database.
-    pub fn create(value: Value) -> Result<Scan, ()> {
-
+    pub async fn create(
+        scan: Scan,
+        connection: &mut SqliteConnection,
+    ) -> Result<Scan, LabwhereError> {
         // TODO: Complete this function.
```

Comment:
We can remove this and replace with some full documentation. This function is now quite complex and could do with some explanation.